src/vs/workbench/contrib/chat/browser/chatEditorActions.ts

```diff
@@ -144,20 +146,32 @@ abstract class AcceptDiscardAction extends Action2 {
 		const chatEditingService = accessor.get(IChatEditingService);
 		const editorService = accessor.get(IEditorService);
 
-		const editor = editorService.activeTextEditorControl;
-		if (!isCodeEditor(editor) || !editor.hasModel()) {
+		const getUri = () => {
+			const notebookEditor = getNotebookEditorFromEditorPane(editorService.activeEditorPane);
+			if (notebookEditor) {
+				return notebookEditor.textModel?.uri;
+			}
+			const editor = editorService.activeTextEditorControl;
+			if (!isCodeEditor(editor) || !editor.hasModel()) {
+				return;
+			}
+			return editor.getModel().uri;
+		};
+
+		const uri = getUri();
```

Comment:
Can we just compute this variable without defining and calling the function? Some good old `let uri: Uri|undefined...` and some if/else